src/main/java/schmidt/core/Schmidt.java

```diff
@@ -0,0 +1,307 @@
+            String command;
+            try {
+                // split input by whitespace and extract the command
+                command = trimmedInput.split("\\s+")[0].toLowerCase();
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printValidCommands();
+                continue;
+            }
+
+            // Switch statement to handle different commands
+            switch (command) {
+            case "bye":
+                printByeMessage();
+                return;
+            case "list":
+                printListHelper(tasks);
+                continue;
+            case "mark":
+                try {
+                    markTaskHelper(tasks, trimmedInput, true);
+                } catch (NumberFormatException e) {
+                    print("The task number should be a number\n" +
+                            "\tmark <task number>");
+                } catch (IndexOutOfBoundsException e) {
+                    print("Task number out of range\n" +
+                            "\tmark <task number>");
+                } catch (SchmidtException e) {
+                    print(e.getMessage());
+                }
+                continue;
+            case "unmark":
+                try {
+                    markTaskHelper(tasks, trimmedInput, false);
+                } catch (NumberFormatException e) {
+                    print("The task number should be a number\n" +
+                            "\tunmark <task number>");
+                } catch (IndexOutOfBoundsException e) {
+                    print("Task number out of range\n" +
+                            "\tunmark <task number>");
+                } catch (SchmidtException e) {
+                    print(e.getMessage());
+                }
+                continue;
+            case "todo":
+                try {
+                    addTodoHelper(tasks, trimmedInput);
+                } catch (StringIndexOutOfBoundsException e) {
+                    print("The description of a todo cannot be empty\n" +
+                            "\ttodo <description>");
+                }
+                continue;
+            case "deadline":
+                try {
+                    addDeadlineHelper(tasks, trimmedInput);
+                } catch (SchmidtException e) {
+                    print(e.getMessage());
+                }
+                continue;
+            case "event":
+                try {
+                    addEventHelper(tasks, trimmedInput);
+                } catch (SchmidtException e) {
+                    print(e.getMessage());
+                }
+                continue;
+            default:
+                printValidCommands();
+            }
+        }
+    }
+
+    /**
+     * This is a function to add lines to print statements
+     */
+    public static void print(String message) {
+        System.out.println("------------------------------------------------------------");
```

Comment:
Maybe you could consider declaring the "----..." as a constant class-level variable